src/Jackett.Common/Indexers/FileListApi.cs

```diff
@@ -0,0 +1,250 @@
+            AddCategoryMapping(6, TorznabCatType.MoviesHD, "Filme 4K");
+            AddCategoryMapping(7, TorznabCatType.XXX, "XXX");
+            AddCategoryMapping(8, TorznabCatType.PC, "Programe");
+            AddCategoryMapping(9, TorznabCatType.PCGames, "Jocuri PC");
+            AddCategoryMapping(10, TorznabCatType.Console, "Jocuri Console");
+            AddCategoryMapping(11, TorznabCatType.Audio, "Audio");
+            AddCategoryMapping(12, TorznabCatType.AudioVideo, "Videoclip");
+            AddCategoryMapping(13, TorznabCatType.TVSport, "Sport");
+            AddCategoryMapping(13, TorznabCatType.TVSport, "Sport");
+            AddCategoryMapping(15, TorznabCatType.TV, "Desene");
+            AddCategoryMapping(16, TorznabCatType.Books, "Docs");
+            AddCategoryMapping(17, TorznabCatType.PC, "Linux");
+            AddCategoryMapping(18, TorznabCatType.Other, "Diverse");
+            AddCategoryMapping(19, TorznabCatType.MoviesForeign, "Filme HD-RO");
+            AddCategoryMapping(20, TorznabCatType.MoviesBluRay, "Filme Blu-Ray");
+            AddCategoryMapping(21, TorznabCatType.TVHD, "Seriale HD");
+            AddCategoryMapping(22, TorznabCatType.PCPhoneOther, "Mobile");
+            AddCategoryMapping(23, TorznabCatType.TVSD, "Seriale SD");
+            AddCategoryMapping(24, TorznabCatType.TVAnime, "Anime");
+            AddCategoryMapping(25, TorznabCatType.Movies3D, "Filme 3D");
+            AddCategoryMapping(26, TorznabCatType.MoviesBluRay, "Filme 4K Blu-Ray");
+            AddCategoryMapping(27, TorznabCatType.TVUHD, "Seriale 4K");
+        }
+
+        public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken configJson)
+        {
+            LoadValuesFromJson(configJson);
+            logger.Info("Testing provider filelist...");
+            var pingResponse = await CallProviderAsync(new TorznabQuery());
+
+            if (pingResponse.StartsWith("{\"error\""))
+            {
+                throw new ExceptionWithConfigData(pingResponse, configData);
+            }
+            else
+            {
+                try
+                {
+                    var json = JArray.Parse(pingResponse);
+                    if (json.Count > 0)
+                    {
+                        IsConfigured = true;
+                        SaveConfig();
+                        return IndexerConfigurationStatus.Completed;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    throw new ExceptionWithConfigData(ex.Message, configData);
+                }
+                return IndexerConfigurationStatus.RequiresTesting;
+            }
+        }
+
+        protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
+        {
+            var releases = new List<ReleaseInfo>();
+            var response = await CallProviderAsync(query);
```

Comment:
instead of doing all the checks in `ApplyConfiguration`, I think you should a small check here:
```
 if (response.StartsWith("{\"error\""))
          throw new ExceptionWithConfigData(response, configData);
```
and you can remove most of the code in `ApplyConfiguration`. take a look here:
https://github.com/Jackett/Jackett/blob/master/src/Jackett.Common/Indexers/SolidTorrents.cs#L65